authorization/v2/uniformResourceIdentifier.go

```diff
@@ -91,9 +91,12 @@ func (uri UniformResourceIdentifier) MarshalJSON() ([]byte, error) {
 
 func (uri *UniformResourceIdentifier) UnmarshalJSON(data []byte) error {
 	uriString := string(data)
-	unquotedURI := uriString[1 : len(uriString)-1]
 
-	parsedURI, err := NewURIFromString(unquotedURI)
+	if len(uriString) > 0 && uriString[0] == '"' && uriString[len(uriString)-1] == '"' {
```

Comment:
I'd say removing the double quotes should not be conditional. `UniformResourceIdentifier.UnmarshalJSON` unmarshalls a JSON string and if a JSON string is not quoted, it's not a JSON string.